### PR TITLE
Fix error strings to follow Go guidelines

### PR DESCRIPTION
**Pull request by slonopotamus**
Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

### CODE REVIEW

File: git/git.go [hunk 1]
```diff
@@ -237,10 +237,10 @@ func LsTree(ref string) (*subprocess.BufferedCmd, error) {
 func ResolveRef(ref string) (*Ref, error) {
 	outp, err := gitNoLFSSimple("rev-parse", ref, "--symbolic-full-name", ref)
 	if err != nil {
-		return nil, fmt.Errorf("Git can't resolve ref: %q", ref)
+		return nil, fmt.Errorf("git can't resolve ref: %q", ref)
 	}
 	if outp == "" {
-		return nil, fmt.Errorf("Git can't resolve ref: %q", ref)
+		return nil, fmt.Errorf("git can't resolve ref: %q", ref)
```
Comment (bk2204) [inline git/git.go]: In these two cases, I think we're referring to the project Git, not the binary `git`, so keeping these upper case is probably better.

File: errors/errors_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func TestContextOnGoErrors(t *testing.T) {
 }
 
 func TestContextOnWrappedErrors(t *testing.T) {
-	err := NewFatalError(errors.New("Go error"))
+	err := NewFatalError(errors.New("go error"))
```
Comment (bk2204) [inline errors/errors_test.go]: I think it was intended in this file for these to refer to the Go language, in which case an upper case first letter would be appropriate here.

File: git/rev_list_scanner_test.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func TestRevListArgs(t *testing.T) {
 
 func TestRevListScannerCallsClose(t *testing.T) {
 	var called uint32
-	err := errors.New("Hello world")
+	err := errors.New("hello world")
```
Comment (bk2204) [inline git/rev_list_scanner_test.go]: I don't have a strong opinion on this one, but it is customary to write "Hello world" with an initial capital.

File: lfsapi/ntlm.go [hunk 4]
```diff
@@ -144,7 +144,7 @@ func ntlmGetCredentials(creds creds.Creds) (*ntmlCredentials, error) {
 
 	splits := strings.Split(username, "\\")
 	if len(splits) != 2 {
-		return nil, fmt.Errorf("Your user name must be of the form DOMAIN\\user. It is currently '%s'", username)
+		return nil, fmt.Errorf("your user name must be of the form DOMAIN\\user. It is currently '%s'", username)
```
Comment (bk2204) [inline lfsapi/ntlm.go]: I think this one and the one above are intended to be sentences, and so should begin with a capital.

File: lfshttp/verbose.go [hunk 5]
```diff
@@ -23,7 +23,7 @@ func (c *Client) traceRequest(req *http.Request) (*tracedRequest, error) {
 
 	body, ok := req.Body.(ReadSeekCloser)
 	if body != nil && !ok {
-		return nil, fmt.Errorf("Request body must implement io.ReadCloser and io.Seeker. Got: %T", body)
+		return nil, fmt.Errorf("request body must implement io.ReadCloser and io.Seeker. Got: %T", body)
```
Comment (bk2204) [inline lfshttp/verbose.go]: Same comment here about the sentences.

File: errors/errors_test.go (hunk 2 above)
Comment (slonopotamus) [reply errors/errors_test.go]: But at the same time this file says `t.Error("go error should not be a fatal error")` even before my changes.

File: git/git.go (hunk 1 above)
Comment (slonopotamus) [reply git/git.go]: OK, reverted.

File: git/rev_list_scanner_test.go (hunk 3 above)
Comment (slonopotamus) [reply git/rev_list_scanner_test.go]: Completely changed error text here

File: lfsapi/ntlm.go (hunk 4 above)
Comment (slonopotamus) [reply lfsapi/ntlm.go]: Ok, reverted

File: lfshttp/verbose.go (hunk 5 above)
Comment (slonopotamus) [reply lfshttp/verbose.go]: Ok, reverted

**Pull request merged**